src/main/daml/Daml/Finance/Holding/Util.daml

```diff
@@ -68,7 +68,7 @@ acquireImpl holding Base.Acquire{newLockers; context; lockType} = do
   whenSome holding.lock \lock -> do
     lock.lockers === newLockers
     lock.lockType === lockType
-    assertMsg "Contract must be locked." $ lock.lockType == Base.Reentrant
+    assertMsg "Contract must not be locked." $ lock.lockType == Base.Reentrant
```

Comment:
I think the error should rather be something like "Cannot lock a non-reentrant lock again" or "Contract must have re-entrant lock"

Reply:
yes, would prefer "Contract must have re-entrant lock"

Reply:
I refactored to make error messages more explicit

Reply:
How about `Contract is locked. Only re-entrant locks may be locked multiple times`? 

Saying the `Contract must have re-entrant lock` is misleading if the caller was expecting the contract to have no locks on the contract or  a `Semaphore` lock was justifiably utilised before this choice was called.

Reply:
I didn't see your refactor when reviewing. I think your refactor is good 👍

Reply:
I agree with Brian's earlier comment, hence the refactoring.

Reply:
looks better, added 2 comments in the code